test/Microsoft.ML.Tests/Transformers/PcaTests.cs

```diff
@@ -14,47 +13,85 @@ namespace Microsoft.ML.Tests.Transformers
 {
     public sealed class PcaTests : TestDataPipeBase
     {
+        private readonly ConsoleEnvironment _env;
+        private readonly string _dataSource;
+        private readonly TextSaver _saver;
+
         public PcaTests(ITestOutputHelper helper)
             : base(helper)
         {
+            _env = new ConsoleEnvironment(seed: 1, conc: 1);
+            _dataSource = GetDataPath("generated_regression_dataset.csv");
+            _saver = new TextSaver(_env, new TextSaver.Arguments { Silent = true, OutputHeader = false });
         }
 
         [Fact]
         public void PcaWorkout()
         {
-            var env = new ConsoleEnvironment(seed: 1, conc: 1);
-            string dataSource = GetDataPath("generated_regression_dataset.csv");
-            var data = TextLoader.CreateReader(env,
-                c => (label: c.LoadFloat(11), features: c.LoadFloat(0, 10)),
+            var data = TextLoader.CreateReader(_env,
+                c => (label: c.LoadFloat(11), weight: c.LoadFloat(0), features: c.LoadFloat(1, 10)),
                 separator: ';', hasHeader: true)
-                .Read(dataSource);
+                .Read(new MultiFileSource(_dataSource));
```

Comment:
>new MultiFileSource(_dataSource) [](start = 22, length = 32)

I would prefer to have string path rather than MultiFileSource

Reply:
All snippets I had seen were using MultiFileSource. Didn't know that string input was an option, which is much cleaner. Will change to string.

---
In reply to: [227046972](https://github.com/dotnet/machinelearning/pull/1333#discussion_r227046972) [](ancestors = 227046972)